### PR TITLE
tf: enable core dumps for VM pods

### PR DESCRIPTION
**Pull request by howardjohn**
I feel terrible about this solution, but its the best I came up with
after hours of work

### CODE REVIEW

File: pkg/envoy/proxy.go [hunk 1]
```diff
@@ -70,6 +70,10 @@ func NewProxy(cfg ProxyConfig) Proxy {
 		args = append(args, "--component-log-level", cfg.ComponentLogLevel)
 	}
 
+	// Explicitly enable core dumps. This may be desirable more often (by default), but for now we only set it in VM tests.
+	if enableEnvoyCoreDump {
+		args = append(args, "--enable-core-dump")
```
Comment (kyessenov) [inline pkg/envoy/proxy.go:75]: Is this off-by-default? Why do we need this flag now?

Comment (howardjohn) [reply pkg/envoy/proxy.go:75]: TBH I have no clue. It works on k8s without the flag, and only works with this crazy setup with the flag...

**Review by ericvn — APPROVED** [review]

File: pkg/envoy/proxy.go (hunk 1 above)
Comment (kyessenov) [reply pkg/envoy/proxy.go:75]: let's follow up, maybe file an issue why this isn't the default

**Review by kyessenov — APPROVED** [review]

**Pull request merged**